userspace/libsinsp/threadinfo.cpp

```diff
@@ -531,8 +531,8 @@ void sinsp_threadinfo::set_user(uint32_t uid)
 	{
 		m_user.uid = uid;
 		m_user.gid = m_group.gid;
-		strlcpy(m_user.name, "<NA>", sizeof(m_user.name));
-		strlcpy(m_user.homedir, "<NA>", sizeof(m_user.homedir));
+		strlcpy(m_user.name, (uid == 0) ? "root" : "<NA>", sizeof(m_user.name));
```

Comment:
Would it make sense to call `getpwuid_r` to obtain user information? Not sure if that might be to expensive / time-consuming due to underlying name service switch configuration...

Reply:
@hbrueckner the diff hid the context: we're right after the use of `sinsp_usergroup_manager`, which already handles all the logic (and some magic) for getting the user data.

Reply:
Thanks @deepskyblue86